code/modules/html_interface/map/interactive_map.dm

```diff
@@ -9,7 +9,13 @@
 "<div id='switches'>\
```

Comment:
```dm 
#define MAPCONTENT \
  "<div id='switches'>\
  <a href=\"javascript:switchTo(0);\">Switch to mini map</a> \
  <a href=\"javascript:switchTo(1);\">Switch to text-based</a> \
  [get_zlevel_ui_js()] \
  </div> \
  <div id=\"uiMapContainer\">\
  <div id=\"uiMap\" unselectable=\"on\"></div></div>\
  <div id=\"textbased\"></div>"

/proc/get_zlevel_ui_js()
   var/list/s = list("Z-Level:")
   for(var/i in 1 to world.maxz)
        s += "<a href='javascript:changezlevels([i]);'>[i]</a>"
   return s.Join()
```
reuse proc elsewhere as needed

although this is all pointless atm as your JS ```changezlevels()``` function is capped at 6